pipelines/main/platforms/upload_linux.yml

```diff
@@ -1,8 +1,12 @@
 steps:
   - label: ":linux: upload ${TRIPLET?}"
     key:   "upload_${TRIPLET?}"
-    # We only upload to S3 if the branch is `master` or `release-*`.
-    if: ((build.branch == "master") || (build.branch =~ /^release-/))
+    # We only upload to S3 if one of the following criteria are true:
+    # 1. The branch is `master`.
+    # 2. The branch is `release-*`.
+    # 3. The build is a tag build AND the tag is `v*`.
+    # 4. The pipeline is `julia-buildkite` AND the build is a pull request build.
+    if: (build.branch == "master") || (build.branch =~ /^release-/) || (build.tag =~ /^v/) || ((pipeline.slug == "julia-buildkite") && (build.pull_request.id != null))
```

Comment:
I'm fine with uploading on `julia-buildkite` unconditionally; e.g. we don't need to check if it's a pull request.

Reply:
Done!